src/app/shared/services/websocket.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs/Subject';
+import { Observable } from 'rxjs/Observable';
+import { Observer } from 'rxjs/Observer';
+
+import { filter, map } from 'rxjs/operators';
+
+import 'rxjs/add/operator/filter';
+import 'rxjs/add/operator/map';
+
+import { Store } from '@ngrx/store';
+import { AppState } from '../../core/state';
+
+import { SetFood } from '../../core/state/food/food.actions';
+
+@Injectable()
+export class WebSocketService {
+    private subject: Subject<MessageEvent>;
+
+    public wsSocket;
+
+    public connect(url): Subject<MessageEvent> {
+        if (!this.subject) {
+            this.subject = this.create(url);
+        }
+
+        return this.subject;
+    }
+
+    public update(key: string, data: any) {
+        this.wsSocket.next({
+            key,
+            data,
+        });
+    }
+
+    constructor(private store: Store<AppState>) {
+        this.wsSocket = this.connect('ws://localhost:3000');
```

Comment:
need to replace with the current host in production. Could put this in the environment variable